Dockerfile.wolfi

```diff
@@ -42,31 +42,47 @@ RUN mkdir -p "$GEM_HOME" && chmod 777 "$GEM_HOME"
 # ------------------------------------------------------------------------------
 # install the application
 
-# TODO we should look at finding a better way to get the application artifact into
-# the container image, since we are adding lots of stuff here that we do not need
-# and may not want (do we need vendor/ after the make install below, for instance?),
-# contributing to increased image size; for now at least we can remove `.git` (below)
-
-# java is the base image's default us
+# java is the base image's default user
 COPY --chown=java:java --chmod=775 . /home/app
 WORKDIR /home/app
 
 # skip jenv/rbenv setup
 ENV IS_DOCKER=1
 
 RUN make clean install
-
-# Clean up build dependencies
-RUN rm -r /root/.m2
+# add more directories and files not to be copied to the runtime image from /home/app
+RUN rm -rf .git .github .idea .devcontainer .buildkite
 
 # ------------------------------------------------------------------------------
-# remove now-unnecessary packages and directories
+# Runtime stage - using the same base image
+FROM docker.elastic.co/wolfi/jdk:openjdk-21.35-r1@sha256:d7ca36452a68f28e4c4683062241e817b548844820a0ffd087451214e61eb188
+
+USER root
+
+# Set environment variables
+ENV JAVA_HOME=/usr/lib/jvm/default-jvm \
+    PATH=/opt/jruby/bin:/usr/local/bundle/bin:$PATH \
+    GEM_HOME=/usr/local/bundle \
+    BUNDLE_SILENCE_ROOT_WARNING=1 \
+    BUNDLE_APP_CONFIG=/usr/local/bundle \
+    IS_DOCKER=1
+
+# Install runtime dependencies
+RUN apk update && apk add --no-cache libcurl-openssl4=~8.12.1
```

Comment:
If we aren't using `curl` any more I think we can also remove `libcurl-openssl4`. If my memory serves me well, this is only here to resolve a vulnerability.

Reply:
Of course, if something isn't needed, shall be removed.
I obviously didn't touch this, I focused only on the reorg and making the image smaller. IMHO such change should be a separate PR.